server/policies/collection.ts

```diff
@@ -101,11 +101,8 @@ allow(User, "updateDocument", Collection, (user, collection) => {
     return false;
   }
 
-  if (!collection.isPrivate && user.isAdmin) {
-    return true;
-  }
-
   if (
+    !collection.isPrivate ||
```

Comment:
Something about this still doesn't feel quite right, this additional OR case is saying that the membership should be considered if the collection is public, irregardless of the `collection.permission`. 

It _technically_ works right now because the only other permission is ReadWrite but I don't think it's neccessary. The only _needed_ change is the removal of the admin exception above